specs/credentialmanagement/index.src.html

```diff
@@ -763,39 +804,50 @@ <h4 id="interfaces-request-options">
   {{CredentialContainer/get()}}, the caller specifies a few parameters in a
   {{CredentialRequest}} dictionary.
 
-  Note: The {{CredentialRequestOptions}} typedef is an extension point. If and
-  when new types of credentials are introduced that require options, their
-  dictionary types will be added to the typedef so they can be passed into the
-  request.
-
   <pre class="idl">
-    dictionary FederatedCredentialRequestOptions {
-      sequence&lt;USVString&gt; providers;
-      sequence&lt;DOMString&gt; protocols;
+    interface CredentialFilterMap {
+      readonly attribute unsigned long size;
+
+      getter any get(DOMString credentialType);
+      setter creator void set(DOMString credentialType, any value);
+      deleter void remove(DOMString type);
```

Comment:
I am learning WebIDL as I go here. My understanding was that dictionaries needed to define their keys up front which defeats the purpose of this whole proposal.